index.js

```diff
@@ -171,6 +181,19 @@ class KeyringController extends EventEmitter {
     this.keyrings = await this.unlockKeyrings(password);
     this.setUnlocked();
     this.fullUpdate();
+
+    return this.encryptedKey;
+  }
+
+  /* 
+    MV3:  Unlock by encrypted key
+  */
+  async submitEncryptedKey(encryptedKey) {
+    this.keyrings = await this.unlockKeyrings(undefined, encryptedKey);
+    this.setUnlocked();
+    this.fullUpdate();
+
+    return this.encryptedKey;
```

Comment:
What's the purpose of this being returned? Is it only for tests? 

If so, there's nothing stopping you from making an assertion on `theReferenceTheTestHas.encryptedKey` instead. It's not a private field.

Reply:
That's fair.